src/foam/nanos/auth/UserPropertyAvailabilityService.js

```diff
@@ -25,16 +25,22 @@ foam.CLASS({
   javaImports: [
     'foam.dao.ArraySink',
     'foam.dao.DAO',
-    'foam.mlang.predicate.Predicate'
+    'foam.mlang.MLang',
+    'foam.mlang.MLang.*',
+    'foam.nanos.auth.User'
   ],
 
   methods: [
     {
       name: 'checkAvailability',
       javaCode: `
         DAO userDAO = (DAO) getX().get("localUserDAO");
-        
-        ArraySink select = (ArraySink) userDAO.inX(x).where((Predicate) predicate).select(new ArraySink());
+
+        if ( ! targetProperty.equals("userName") && ! targetProperty.equals("email") ) {
+          throw new RuntimeException("Unsupported use of UserPropertyAvailabilityService.");
```

Comment:
Throw 
```suggestion
 throw new AuthorizationException(); 
```
This will show up as 'Permission denied', which is enough.